src/main/storm/mesos/MesosNimbus.java

```diff
@@ -114,94 +120,228 @@ public String getHostName(Map<String, SupervisorDetails> map, String nodeId) {
+    Boolean preferReservedResources = (Boolean) conf.get(CONF_MESOS_PREFER_RESERVED_RESOURCES);
+    if (preferReservedResources != null) {
+      _preferReservedResources = preferReservedResources;
+    }
+    _container = Optional.fromNullable((String) conf.get(CONF_MESOS_CONTAINER_DOCKER_IMAGE));
+    _scheduler = new NimbusScheduler(this);
+    createLocalServerPort();
+    setupHttpServer();
+
+    _conf.put(Config.NIMBUS_HOST, _configUrl.getHost());
+    // Generate YAML to be served up to clients
+    Path generatedConfPath = Paths.get("generated-conf");
+    if (!generatedConfPath.toFile().exists() && !generatedConfPath.toFile().mkdirs()) {
+      throw new RuntimeException("Couldn't greate generated-conf dir");
+    }
+    File generatedConf = Paths.get(generatedConfPath.toString(), "storm.yaml").toFile();
+    Yaml yaml = new Yaml();
+    FileWriter writer = new FileWriter(generatedConf);
+    yaml.dump(_conf, writer);
+  }
+
+
+  public void doRegistration(final SchedulerDriver driver, Protos.FrameworkID id) {
+    _driver = driver;
+    try {
+      _state.put(FRAMEWORK_ID, id.getValue());
+    } catch (IOException e) {
+      LOG.error("Halting process...", e);
+      Runtime.getRuntime().halt(1);
+    }
+    Number filterSeconds = Optional.fromNullable((Number) _conf.get(CONF_MESOS_OFFER_FILTER_SECONDS)).or(120);
+    final Protos.Filters filters = Protos.Filters.newBuilder()
+        .setRefuseSeconds(filterSeconds.intValue())
+        .build();
+    _offers = new RotatingMap<>(
+        new RotatingMap.ExpiredCallback<Protos.OfferID, Protos.Offer>() {
+          @Override
+          public void expire(Protos.OfferID key, Protos.Offer val) {
+            driver.declineOffer(
+                val.getId(),
+                filters
+            );
+          }
+        }
+    );
+
+    Number lruCacheSize = Optional.fromNullable((Number) _conf.get(CONF_MESOS_OFFER_LRU_CACHE_SIZE)).or(1000);
+    final int intLruCacheSize = lruCacheSize.intValue();
+    _usedOffers = Collections.synchronizedMap(new LinkedHashMap<Protos.TaskID, Protos.Offer>(intLruCacheSize + 1, .75F, true) {
+      // This method is called just after a new entry has been added
+      public boolean removeEldestEntry(Map.Entry eldest) {
+        return size() > intLruCacheSize;
+      }
+    });
+
+    Number offerExpired = Optional.fromNullable((Number) _conf.get(Config.NIMBUS_MONITOR_FREQ_SECS)).or(60);
+    Number expiryMultiplier = Optional.fromNullable((Number) _conf.get(CONF_MESOS_OFFER_EXPIRY_MULTIPLIER)).or(2.5);
+    _timer.scheduleAtFixedRate(new TimerTask() {
+      @Override
+      public void run() {
+        try {
+          synchronized (_offersLock) {
+            _offers.rotate();
+          }
+        } catch (Throwable t) {
+          LOG.error("Received fatal error Halting process...", t);
+          Runtime.getRuntime().halt(2);
+        }
+      }
+    }, 0, Math.round(1000 * expiryMultiplier.doubleValue() * offerExpired.intValue()));
```

Comment:
ok, so this slightly changes the previous timing.  It used to be 2000*10 (by default) == 20 seconds.  It's not 1000*2.5*10 == 25 seconds.  I think that's ok, just calling it out.